impl/src/main/java/org/jboss/weld/context/beanstore/AttributeBeanStore.java

```diff
@@ -104,6 +101,20 @@ public boolean attach() {
         }
     }
 
+    /**
+     * Read all relevant attributes from the backing store and copy instances which are not present in the local bean store.
+     */
+    public void readAttributes() {
```

Comment:
I suggest using a different name that would better explain that this method loads entries from the backing store to the local store.